### DISCUSSION TITLE
TSX Support on AVA

### DISCUSSION THREAD

**Discussion by Abiji-2020** [avajs/ava#3351] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/98341c6d-eb8b-4878-9f32-9ba7788aa0ce)

while trying to run a test file on tsx using ava, it shows unknown file extension. 

my ava configuration is 

```


const processOnly = new Set([
	"--import=tsimp"
]);
export default {
    extensions: {
        js:true,
        ts: 'module',
        tsx: 'module'
    },
    filterNodeArgumentsForWorkerThreads: argument => !processOnly.has(argument)
}
```

**Comment by novemberborn** [comment, marked as answer]
I'm not sure if `filterNodeArgumentsForWorkerThreads` is needed or what it's meant to do, I don't see you passing the `--import=tsimp` argument.

The error looks like the code is importing from ` .tsx` file and that's not a recognized extension, which suggests `tsimp` isn't active. But I don't know if `tsimp` supports `.tsx`.

**Reply by Abiji-2020** [reply]
yeah, the problem was i wasn't passing the `tsimp`, after passing it, the test started working.
